sklearn/kernel_ridge.py

```diff
@@ -129,6 +133,18 @@ class KernelRidge(MultiOutputMixin, RegressorMixin, BaseEstimator):
     KernelRidge(alpha=1.0)
     """
 
+    _parameter_constraints = {
+        "alpha": [Interval(Real, 0, None, closed="left"), np.ndarray],
+        "kernel": [
+            StrOptions(set(PAIRWISE_KERNEL_FUNCTIONS.keys()) | {"precomputed"}),
+            callable,
+        ],
+        "gamma": [Interval(Real, 0, None, closed="left"), None],
+        "degree": [Interval(Real, 0, None, closed="left")],
```

Comment:
We should update the documentation above as well because it states `float`.

```suggestion
        "degree": [Interval(Integral, 0, None, closed="left")],
```

Reply:
Got it, made the change in docstring as well for `degree` parameter.